### ISSUE TITLE
wpt.fyi status checks are not showing up on WPT repo

### ISSUE THREAD

**Issue by stephenmcgruer** [web-platform-tests/wpt.fyi#1660]
See https://github.com/web-platform-tests/wpt/pull/20386#issuecomment-557469514

cc @foolip @Hexcles 

Investigating now. I'm seeing 500 errors on `/api/webhook/check` for prod wpt.fyi, which may be related:

`2019-11-22 04:42:08.264 PST payload signature check failed`

**Comment by stephenmcgruer** [comment]
I would hazard a guess that the error comes from https://github.com/google/go-github/blob/master/github/messages.go#L201

**Comment by stephenmcgruer** [comment]
My suspicion would be that when the app was changed for #1557 , this changed the secret in some way, but I'm not familiar with the flow there so that may be a red herring.

**Comment by stephenmcgruer** [comment]
We read the secret from https://github.com/web-platform-tests/wpt.fyi/blob/c114bd7f665df3c2ab944d21691df810868d6183/api/checks/webhook.go#L53:

`	secret, err := shared.GetSecret(ds, "github-check-webhook-secret")`

Which reads it from the datastore.

**Comment by stephenmcgruer** [comment]
The secret at https://github.com/organizations/web-platform-tests/settings/apps/wpt-fyi and the one stored in the datastore appear to be different. I'm going to try backing up the datastore entry, changing it, and seeing if that helps.

**Comment by stephenmcgruer** [comment]
That does not appear to have helped. I am now suspecting that https://github.com/organizations/web-platform-tests/settings/apps/wpt-fyi is not the right secret; so much for the quick fix. Now to actually read https://developer.github.com/webhooks/securing/#validating-payloads-from-github and see how this works :)

**Comment by stephenmcgruer** [comment]
For now, I have restored the previous secret in the datastore.

**Comment by stephenmcgruer** [comment]
For the life of me, I cannot figure out **who** actually sends requests to `/api/webhook/check`... it doesn't seem to be documented anywhere.

**Comment by stephenmcgruer** [comment]
Ok, there's a webhook secret stored in https://github.com/organizations/web-platform-tests/settings/apps/wpt-fyi that I missed. Resetting that to the one in the datastore.

**Comment by stephenmcgruer** [comment]
Also, I think we've found the cause. Chrome password manager is overriding bits of the settings page, which means that when @Hexcles changed the app name his password manager probably reset the secret too.

**Comment by stephenmcgruer** [comment]
From the logs side, things are recovering. Looking for a WPT PR that has the checks showing up.

**Comment by foolip** [comment]
Thanks @stephenmcgruer for tackling this today!

**Comment by stephenmcgruer** [comment]
I believe we're good here, e.g. see https://github.com/web-platform-tests/wpt/pull/20402

Postmortem is well underway too.

**Comment by stephenmcgruer** [comment]
This happened **again** (see https://github.com/web-platform-tests/wpt/pull/20418#issuecomment-557908941). I checked the app settings page, and for some reason the `value` of the webhook secret input was `value='value=[secret]'`... I've reset it to just `value=[secret]` and expect that will fix the 500s.

@Hexcles were you editing the wpt.fyi app again?

Also @Hexcles - did our alert trigger? /api/webhook/check was 500-ing like crazy...

**Comment by Hexcles** [comment]
No I didn't... That's really weird. I'd really hope there's an auditing
log.

**Comment by Hexcles** [comment]
> Also @Hexcles - did our alert trigger? /api/webhook/check was 500-ing like crazy...

Yes, but apparently Stackdriver failed to send emails to group aliases. Would you mind me adding your email directly, along with mine?

**Comment by stephenmcgruer** [comment]
Sgtm. If we are worried about the corp filter blocking it, also happy to add my chromium or even personal email...

**Comment by Hexcles** [comment]
I might've got an explanation of what happened: I forgot to save the setting when trying to fix my screw-up last Friday -- remember I told you I accidentally did it again last Friday @stephenmcgruer ? I told you I fixed it, but looking at the open tabs of my workstation, I realized the settings tab was still open so chances are I did not actually click save... Sorry about that.

On the bright side, this has been a good test of our alerting setup -- and it didn't work.

**Comment by Hexcles** [comment]
There is an audit log for admin activities, but somehow it doesn't include changes to apps... Sigh.

https://github.com/organizations/web-platform-tests/settings/audit-log

**Comment by stephenmcgruer** [comment]
Oh dear :(. These things happen though, and if I had more control over the app settings I would definitely have included changing that UI as part of the postmortem (adding a confirmation box, for one!).

Now that we seem to have alerting working (I got an email saying the outage was over); would we have continued to receive emails if we thought we had fixed it but it was still actually broken? (I.e. are there 'update' emails whilst an outage still continues?)
